2-frontend/src/components/visual/Pagination/Pagination.jsx

```diff
@@ -0,0 +1,80 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { canGoBack, canGoForward } from "./paginationUtils";
+import './style.scss';
+
+const Pagination = ({
+  totalItems,
+  page,
+  offset,
+  onPageChange
+}) => {
+  const [currentPage, setCurrentPage] = useState(0);
+
+  // If receiving a specific page from the parent, set that page
+  if (page !== currentPage) {
+    setCurrentPage(page);
+  }
+
+  const handlePageChange = jump => {
+    setCurrentPage(prev => {
+      if (jump === 1 && !canGoForward(prev, totalItems, offset)) {
+        return prev;
+      }
+      else if (jump === -1 && !canGoBack(prev)) {
+        return prev;
+      }
+      const nextPage = prev + jump;
+      onPageChange(nextPage);
+      return nextPage;
+    });
+  };
+
+  const totalPages = totalItems > 0 ? Math.ceil(totalItems / offset) : 1;
+
+  return (
+    <div className="pagination-container">
+      {
+        canGoBack(currentPage) && (
+          <button id="prevButton" onClick={() => handlePageChange(-1)}>Previous</button>
+        )
+      }
+      {totalItems > 0 && 
+        <span id="totalItems" className="pagination-text">{page + 1} of {totalPages}</span>
+      }
+      {
+        canGoForward(currentPage, totalItems, offset) && (
+          <button id="nextButton" onClick={() => handlePageChange(1)}>Next</button>
+        )
+      }
+    </div>
+  );
+};
+
+Pagination.propTypes = {
```

Comment:
I think `PropTypes` are quite useful when dealing with plain JS, allowing us to add documentation about the component API and it also gives some insights while debugging. 
They may overlap a bit, but Typescript purpose is towards developing and producing code, since it gives us validation in compile time, while PropTypes perform validation in runtime.
I don't have experience in Typescript, at least no more than a few tutorials that I found. But mainly for larger projects, I think that TS would have an advantage over plain JS, from the stability that it can bring into the code.